prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/FhirPathUtils.kt

```diff
@@ -87,6 +95,40 @@ object FhirPathUtils : Logging {
         bundle: Bundle,
         expressionNode: ExpressionNode
     ): String {
-        return defaultPathEngine.evaluateToString(appContext, focusResource, bundle, bundle, expressionNode)
+        val evaluated = defaultPathEngine.evaluate(appContext, focusResource, bundle, bundle, expressionNode)
+        if (evaluated.size == 0) return ""
+        if (evaluated.size > 1 || !evaluated[0].isPrimitive) {
+            // Should only be evaluating primitives here
+            throw IllegalStateException()
+        }
+        return when (val primitive = evaluated[0]) {
+            is InstantType, is DateTimeType -> convertDateTimeToHL7(primitive)
+            is DateType -> convertDateToHL7(primitive)
+            is TimeType -> convertTimeToHL7(primitive)
+            else -> defaultPathEngine.convertToString(primitive)
+        }
+    }
+    fun convertDateTimeToHL7(dateTime: Base): String {
+        val pattern = "YYYYMMddHHmmss.SSSSxxxx"
+        val formatter: DateTimeFormatter = DateTimeFormatter.ofPattern(pattern)
```

Comment:
Predefine these formatters in a companion object.  This will save time and increase performance.  If there is very little code left on each you may want to consider if it is worth including any code left on these functions to the one above.

Reply:
Can we talk about the compilation benefit of the companion object? I'm not sure I understand what you're asking for here.

Reply:
When you have the code creating a DateTimeFormatter like here, an object of that type will be created every single time this method is called.  The object will be the same and do the same every time, so you could refactor that out for reused.  You can add to the class this:
```
companion object {
   private val dateTimeFormatter = DateTimeFormatter.ofPattern("YYYYMMddHHmmss.SSSSxxxx")
}
```
this will create this static object once for the life of the JVM and you can reuse it anywhere.  This is a minor optimization, but one that can create this formatter as a constant for your class and define the pattern at a higher level in your class (which is good for these type of constants).

Reply:
Gotcha - I think whats confusing is that the FHIRPathUtils is already an object and we can't create companion objects inside of other objects. Is it worth switching FHIRPathUtils to a class to allow for this optimization?